core/src/main/scala/gnieh/diffson/package.scala

```diff
@@ -15,6 +15,43 @@
 */
 package gnieh
 
+import scala.collection.immutable.Queue
+
 /** This package contains an implementation of Json JsonPatch, according to [RFC-6902](http://tools.ietf.org/html/rfc6902)
  */
-package object diffson
+package object diffson {
+
+  private val IsDigit = "(0|[1-9][0-9]*)".r
+
+  type Pointer = Queue[Either[String, Int]]
+
+  implicit class PointerOps(val pointer: Pointer) extends AnyVal {
```

Comment:
Yes that was the reason why I separated both. But a Json pointer outside `JsonSupport` does not make sense as it is something that points to a concrete Json element. This separation was made at the beginning when diffson only had support for spray-json. I should have moved it when I added support for several Json backends.